### PR TITLE
Optimized version of the 'linear' mode of the operation Interpolate-4

### PR DESCRIPTION
**Pull request by avoskoboinyk-lohika**
### Details:
- Optimize reference kernel of the 'linear' mode
- Add JIT kernels for 'linear' mode

### Tickets:
 - *89267*

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@chenhu-wang I guess we need your review first

**Comment by avoskoboinyk-lohika** [comment]
Now it depends on
https://github.com/openvinotoolkit/oneDNN/pull/143
https://github.com/openvinotoolkit/oneDNN/pull/144

before being ready to be merged.

**Comment by dmitry-gorokhov** [comment]
> Now it depends on [openvinotoolkit/oneDNN#143](https://github.com/openvinotoolkit/oneDNN/pull/143) [openvinotoolkit/oneDNN#144](https://github.com/openvinotoolkit/oneDNN/pull/144)
> 
> before being ready to be merged.

Is it possible to unite them into one PR? This will simplify merge

File: src/common/transformations/src/transformations/op_conversions/convert_interpolate1_to_interpolate4.cpp [hunk 1]
```diff
@@ -44,20 +44,7 @@ ngraph::pass::ConvertInterpolate1ToInterpolate4::ConvertInterpolate1ToInterpolat
...
-            // TODO: delete this conditional statement, when CPU will have
-            // optimized version of the 'linear' mode.
-            if (input_shape_rank < 5) {
-                attrsV4.mode = ngraph::op::v4::Interpolate::InterpolateMode::LINEAR_ONNX;
-            } else if (input_shape_rank == 5) {
-                attrsV4.mode = ngraph::op::v4::Interpolate::InterpolateMode::LINEAR;
-            } else {
-                return false;
-            }
+            attrsV4.mode = ngraph::op::v4::Interpolate::InterpolateMode::LINEAR;
```
Comment (dmitry-gorokhov) [inline src/common/transformations/src/transformations/op_conversions/convert_interpolate1_to_interpolate4.cpp:47]: We have to make sure that newly optimized version works at least not slower than LINEAR_ONNX.
@chenhu-wang May I ask you to help here with proper benchmarking?

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp [hunk 2]
```diff
@@ -366,135 +441,1017 @@ struct jit_uni_interpolate_kernel_f32 : public jit_uni_interpolate_kernel, publi
...
+                    mov(reg_tmp_32, 1);
+                    kmovw(k_mask, reg_tmp_32);
+
+                    vpblendmd(vmm_index_saved_1 | k_mask, vmm_index_saved_1, vmm_tmp);
+                    vblendmps(vmm_weights_saved_1 | k_mask, vmm_weights_saved_1, vmm_weights);
+                } else {
+                    if (isa == cpu::x64::avx2) {
+                        vpblendd(vmm_index_saved_1, vmm_index_saved_1, vmm_tmp, 1);
+                    } else {
+                        uni_vpblendw(vmm_index_saved_1, vmm_index_saved_1, vmm_tmp, 3);
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp:1105]: Why we need to add new uni_* mnemonic here given it is used only for SSE code path?

Comment (avoskoboinyk-lohika) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp:1105]: Not only for SSE, but for AVX too.

File: src/common/transformations/src/transformations/op_conversions/convert_interpolate1_to_interpolate4.cpp (hunk 1 above)
Comment (chenhu-wang) [reply src/common/transformations/src/transformations/op_conversions/convert_interpolate1_to_interpolate4.cpp:47]: @dmitry-gorokhov  talked online meeting, will do this.

**Comment by akladiev** [comment]
This PR will be closed in 2 weeks in case of no activity.

**Comment by akladiev** [comment]
This PR was closed because it has been stalled for 2 week with no activity.

**Pull request closed without merging**